### PR TITLE
Add simple balance endpoint (for amm usage)

### PR DESCRIPTION
**Pull request by diamondhands0**
There wasn't a clean way to get the balance of a set of coins, which is needed in order to determine if the amm should initialize a market. Added this endpoint to address this need.

There's also a thorough test for it in the amm code here:
https://github.com/deso-protocol/amm-service/blob/f4953a8f1a141110a10f2ced44e7983176d5d46e/ammlib/amm_config_test.go#L1170

### CODE REVIEW

File: routes/server.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ const (
 	RoutePathGetSingleProfile                           = "/api/v0/get-single-profile"
 	RoutePathGetSingleProfilePicture                    = "/api/v0/get-single-profile-picture"
 	RoutePathGetHodlersForPublicKey                     = "/api/v0/get-hodlers-for-public-key"
+	RoutePathGetBalancesForPublicKey                    = "/api/v0/get-balances-for-public-key"
```
Comment (lazynina) [inline routes/server.go:80]:
```suggestion
	RoutePathGetTokenBalancesForPublicKey                    = "/api/v0/get-token-balances-for-public-key"
```
Can we make it more clear that this is for DeSo Token (FKA DAO coin) balances?

File: routes/server.go [hunk 2]
```diff
@@ -929,6 +930,13 @@ func (fes *APIServer) NewRouter() *muxtrace.Router {
 			fes.GetHodlersForPublicKey,
 			PublicAccess,
 		},
+		{
+			"GetBalancesForPublicKey",
+			[]string{"POST", "OPTIONS"},
+			RoutePathGetBalancesForPublicKey,
+			fes.GetBalancesForPublicKey,
```
Comment (lazynina) [inline routes/server.go:937]:
```suggestion
			"GetTokenBalancesForPublicKey",
			[]string{"POST", "OPTIONS"},
			RoutePathGetTokenBalancesForPublicKey,
			fes.GetTokenBalancesForPublicKey,
```

File: routes/user.go [hunk 3]
```diff
@@ -1479,6 +1479,119 @@ func (fes *APIServer) GetHodlersForPublicKey(ww http.ResponseWriter, req *http.R
...
+	// A base-10 encoded string of the balance in base units, which is internally
+	// represented as a uint256
+	BalanceBaseUnits string
+}
+
+type GetBalancesForPublicKeyResponse struct {
+	Balances map[string]*SimpleBalanceResponse
+}
+
+func (fes *APIServer) GetBalancesForPublicKey(ww http.ResponseWriter, req *http.Request) {
```
Comment (lazynina) [inline routes/user.go:1511]:
```suggestion
func (fes *APIServer) GetTokenBalancesForPublicKey(ww http.ResponseWriter, req *http.Request) {
```

File: routes/user.go [hunk 4]
```diff
@@ -1479,6 +1479,119 @@ func (fes *APIServer) GetHodlersForPublicKey(ww http.ResponseWriter, req *http.R
...
+	}
+
+	// Decode the public key for which we are fetching balances
+	if requestData.UserPublicKey == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("GetBalancesForPublicKey: Missing UserPublicKey"))
+		return
+	}
+	userPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.UserPublicKey)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetHodlersForPublicKey: Problem decoding user public key: %v", err))
```
Comment (lazynina) [inline routes/user.go:1538]: Clean up error prefixes

File: routes/user.go [hunk 5]
```diff
@@ -1479,6 +1479,119 @@ func (fes *APIServer) GetHodlersForPublicKey(ww http.ResponseWriter, req *http.R
...
+	// the balance of for the user. For DESO,
+	// can specify either "DESO" or the ZeroPkid via base58check encoding.
+	// Either will work.
+	CreatorPublicKeys []string
+
+	// If unset, defaults to TxnStatusInMempool
+	TxnStatus TxnStatus `safeForLogging:"true"`
+}
+
+type SimpleBalanceResponse struct {
```
Comment (lazynina) [inline routes/user.go:1496]:
```suggestion
type SimpleTokenBalanceResponse struct {
```

File: routes/user.go [hunk 6]
```diff
@@ -1479,6 +1479,119 @@ func (fes *APIServer) GetHodlersForPublicKey(ww http.ResponseWriter, req *http.R
...
+		balanceEntry, _, _ := utxoView.GetBalanceEntryForHODLerPubKeyAndCreatorPubKey(
+			userPublicKeyBytes, creatorPkBytes, true)
+		if balanceEntry == nil || balanceEntry.IsDeleted() {
+			balanceEntry = &lib.BalanceEntry{}
+		}
+		// Convert balanceEntry uint256 to string
+		balancesMap[creatorPublicKeyStr] = &SimpleBalanceResponse{
+			UserPublicKeyBase58Check:    requestData.UserPublicKey,
+			CreatorPublicKeyBase58Check: creatorPublicKeyStr,
+			BalanceBaseUnits:            balanceEntry.BalanceNanos.String(),
```
Comment (lazynina) [inline routes/user.go:1581]: Would returning the hex string here also be useful? So we could do `BalanceBaseUnits` and `BalanceBaseUnitsHex` or something. Not sure if you're concerned about losing precision when going to base10 string.

**Review by lazynina — CHANGES_REQUESTED** [review]

File: routes/server.go (hunk 1 above)
Comment (diamondhands0) [reply routes/server.go:80]: Fixed.

File: routes/server.go (hunk 2 above)
Comment (diamondhands0) [reply routes/server.go:937]: Fixed.

File: routes/user.go (hunk 3 above)
Comment (diamondhands0) [reply routes/user.go:1511]: Fixed.

File: routes/user.go (hunk 4 above)
Comment (diamondhands0) [reply routes/user.go:1538]: Fixed.

File: routes/user.go (hunk 5 above)
Comment (diamondhands0) [reply routes/user.go:1496]: Fixed.

File: routes/user.go (hunk 6 above)
Comment (diamondhands0) [reply routes/user.go:1581]: We don't lose precision when we convert to a base-10 string. It's the same value, just easier to read.

**Review by lazynina — APPROVED** [review]

**Pull request merged**